WooCommerce/NotificationExtension/OrderNotificationDataService.swift

```diff
@@ -36,6 +36,19 @@ final class OrderNotificationDataService {
         notesRemote = NotificationsRemote(network: network)
     }
 
+    @MainActor
+    func loadOrderFrom(notification: PushNotification) async throws -> (Note, Order) {
+        let note: Note
+        if let notificationNote = notification.note {
+            note = notificationNote
+        }
+        else {
+            note = try await loadNotification(noteID: notification.noteID)
+        }
```

Comment:
Tip: For these kind of code I'd like to do something like 

```swift
let note: Note = try await {
    if let notificationNote = notification.note {
        return notificationNote
    } else {
        return try await loadNotification(noteID: notification.noteID)
    }
}()
```

In order to make it clearer for the reader that the if/else is just for the note creation and should not have any other side effect